_gsocproposals/2022/proposal_FastSim-pipeline-optimization.md

```diff
@@ -0,0 +1,54 @@
+Machine Learning (ML) techniques such as generative modeling are used as fast simulation alternatives to learn to generate showers in a calorimeter, i.e. simulating the calorimeter response to certain particles. The pipeline of a fast simulation solution can be categorized into five components: data preprocessing, ML model design, validation, inference and optimization. The preprocessing module allows us to derive a suitable representation of showers, to perform data cleaning, scaling and encoding. The preprocessed data is then used by the generative model for training. In order to search for the best set of hyperparameters of the model, techniques such as Automatic Machine Learning (AutoML) are used. The validation component is based on comparing different ML metrics and physics quantities between the input and generated data. After training and validation the model is converted into a format that can be used for inference in C++. This allows its application directly in the frameworks of physics experiments. The optimization component is used to further reduce the memory footprint of the model at inference time. Moreover, optimization techniques are also used at training time to reduce the number of trainable parameters. 
+The aim of this project is to optimize the ML pipeline of the fast simulation approach using the open-source platform [Kubeflow](https://github.com/kubeflow/kubeflow). Furthermore, a byproduct of this project is that the student will gain expertise in cutting-edge ML techniques, and learn to use them in the context of high granularity image generation and fast simulation. Moreover, this project can serve as a baseline for future ML pipelines for all experiments at CERN. 
+
+
+## Project Milestones
+
+* Build Kubeflow pipeline from the existing project
+* Evaluate and study each pipeline component 
+* If time allows, benchmark optimization techniques to reduce the memory footprint of the model.
+
+## Expected results
+
+An efficient ML pipeline with documentation that must include:
+
+* Comparison and evaluation of the pipeline with all its components
+* Pipeline documentation 
+* Plots and outputs of Kubeflow performance monitoring
+* If time allows, comparison and evaluation of memory footprint optimization techniques
+
+## Requirements
+
+* Solid knowledge of ML
+* Strong python skills
+
+## Evaluation tasks
+
+Python and ML exercises.
+
+## Mentors
+
+* [Dalila Salamani](mailto:dalila.salamani@cern.ch)
+* [Anna Zaborowska](mailto:anna.zaborowska@cern.ch)
```

Comment:
The main mentor should be emphasized and the affiliation should be present for all mentors, e.g. `**[Dalila Salamani](mailto:dalila.salamani@cern.ch)** (CERN)`

Reply:
fixed, thanks!